include/jwt-cpp/jwt.h

```diff
@@ -99,6 +99,206 @@ namespace jwt {
+			ecdsa_do_sign_failed,
+			digestinit_failed,
+			digestupdate_failed,
+			digestfinal_failed,
+			rsa_padding_failed,
+			rsa_private_encrypt_failed
+		};
+		/**
+		 * \brief Errorcategory for signature generation errors
+		 */
+		inline std::error_category& signature_generation_error_category() {
+			class signature_generation_error_cat : public std::error_category
+			{
+			public:
+				const char* name() const noexcept override { return "signature_generation_error"; };
+				std::string message(int ev) const override {
+					switch(static_cast<signature_generation_error>(ev)) {
+					case signature_generation_error::ok: return "no error";
+					case signature_generation_error::hmac_failed: return "hmac failed";
+					case signature_generation_error::create_context_failed: return "failed to create signature: could not create context";
+					case signature_generation_error::signinit_failed: return "failed to create signature: SignInit failed";
+					case signature_generation_error::signupdate_failed: return "failed to create signature: SignUpdate failed";
+					case signature_generation_error::signfinal_failed: return "failed to create signature: SignFinal failed";
+					case signature_generation_error::ecdsa_do_sign_failed: return "failed to generate ecdsa signature";
+					case signature_generation_error::digestinit_failed: return "failed to create signature: DigestInit failed";
+					case signature_generation_error::digestupdate_failed: return "failed to create signature: DigestUpdate failed";
+					case signature_generation_error::digestfinal_failed: return "failed to create signature: DigestFinal failed";
+					case signature_generation_error::rsa_padding_failed: return "failed to create signature: RSA_padding_add_PKCS1_PSS_mgf1 failed";
+					case signature_generation_error::rsa_private_encrypt_failed: return "failed to create signature: RSA_private_encrypt failed";
+					default: return "unknown error code";
+					}
+				}
+			};
+			static signature_generation_error_cat cat = {};
+			return cat;
+		}
+
+		inline std::error_code make_error_code(signature_generation_error e) {
+			return {static_cast<int>(e), signature_generation_error_category()};
+		}
+
+		enum class token_verification_error {
+			ok = 0,
+			wrong_algorithm,
+			missing_claim,
+			claim_type_missmatch,
+			claim_value_missmatch,
+			token_expired,
+			audience_missmatch
+		};
+		inline std::error_category& token_verification_error_category() {
+			class token_verification_error_cat : public std::error_category
+			{
+			public:
+				const char* name() const noexcept override { return "token_verification_error"; };
+				std::string message(int ev) const override {
+					switch(static_cast<token_verification_error>(ev)) {
+					case token_verification_error::ok: return "no error";
+					case token_verification_error::wrong_algorithm: return "wrong algorithm";
+					case token_verification_error::missing_claim: return "decoded_jwt is missing required claim";
```

Comment:
```suggestion
					case token_verification_error::missing_claim: return "decoded JWT is missing required claim(s)";
```

I noticed the other error codes return "failed to ...." this one does not, in case you want them too.